tasks.py

```diff
@@ -366,22 +368,36 @@ def ci_docs_add_ssh(c, ssh_key, host_key):
 
 
 def get_sanitizers():
-    sanitizers = os.getenv('SANITIZERS')
+    sanitizers = os.getenv('SANITIZERS', '')
     if sanitizers != '':
         sanitizers = sanitizers.split('+')
         sanitizers = ['--enable-' + sanitizer for sanitizer in sanitizers]
         sanitizers = ' '.join(sanitizers)
     return sanitizers
 
+def get_c_compiler():
+    if os.getenv('COMPILER', 'clang') == 'clang':
+        return f'clang-{clang_version}'
+    return 'gcc'
+
+def get_cxx_compiler():
+    if os.getenv('COMPILER', 'clang') == 'clang':
+        return f'clang++-{clang_version}'
+    return 'g++'
+
+def get_optimizations():
+    if os.getenv('OPTIMIZATIONS', 'yes') == 'yes':
+        return "-O1"
+    return "-O0"
 
 def get_cflags():
     return " ".join([
-        "-O1",
+        get_optimizations(),
         "-Werror=vla",
         "-Werror=shadow",
         "-Wformat=2",
         "-Werror=format-security",
-        "-Werror=string-plus-int",
+        "-Werror=string-plus-int" if os.getenv('COMPILER', 'clang') == 'clang' else '',
```

Comment:
perhaps use `get_c_compiler()` here?

Reply:
Right. Something like `get_c_compiler().startswith('clang')`.

Reply:
That does not look better to me than the current version, as it would introduce a new abstraction and immediately make an assumption about the value returned by that abstraction.